src/Analyzers/CSharp/CodeFixes/UseExplicitArrayInExpressionTree/CSharpUseExplicitArrayInExpressionTreeCodeFixProvider.cs

```diff
@@ -0,0 +1,179 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseExplicitArrayInExpressionTree;
+
+using static SyntaxFactory;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseExplicitArrayInExpressionTree), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed partial class CSharpUseExplicitArrayInExpressionTreeCodeFixProvider() : SyntaxEditorBasedCodeFixProvider
+{
+    private const string CS9226 = nameof(CS9226); // An expression tree may not contain an expanded form of non-array params collection parameter.
+
+    public override ImmutableArray<string> FixableDiagnosticIds => [CS9226];
+
+    private static (int index, ITypeSymbol arrayElementType)? DetermineFirstArgumentIndexToWrap(
+        SemanticModel semanticModel, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken)
+    {
+        // Only offer if we actually know which method this was calling.
+        if (semanticModel.GetSymbolInfo(invocationExpression, cancellationToken).Symbol is not IMethodSymbol originalMethod ||
+            originalMethod.Parameters is not [.., { IsParams: true } originalParamsParameter])
+        {
+            return null;
+        }
+
+        // Now we have to find a sibling method that exists that is the same, except that it takes a params array instead.
+        var memberGroup = semanticModel.GetMemberGroup(invocationExpression.Expression, cancellationToken);
+
+        foreach (var symbol in memberGroup)
+        {
+            if (symbol is IMethodSymbol currentMethod &&
+                currentMethod.Parameters.Length == originalMethod.Parameters.Length &&
+                currentMethod.Parameters is [.., { IsParams: true, Type: IArrayTypeSymbol arrayType }])
+            {
+                if (currentMethod.ReturnsVoid != originalMethod.ReturnsVoid)
+                    continue;
+
+                // Different return type, can't switch to this method.
+                if (!currentMethod.ReturnsVoid && !Equals(currentMethod.ReturnType, originalMethod.ReturnType))
+                    continue;
+
+                for (int i = 0, n = currentMethod.Parameters.Length - 1; i < n; i++)
+                {
+                    var originalParameter = originalMethod.Parameters[i];
+                    var currentParameter = currentMethod.Parameters[i];
+
+                    // Different parameter types, look for another sibling that matches
+                    if (!originalParameter.Type.Equals(currentParameter.Type))
+                        continue;
```

Comment:
>continue; [](http://example.com/codeflow?start=24&length=9)

is this intending to continue on the outer loop?

Reply:
yup.  will fix.